### DISCUSSION TITLE
Run After Application Shell Initialization

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#10339] [General] [answered]
Hi,
I'm trying to change a statusbar element based on ApplicationShell's bottom active widget to do so, I need to access application shell on startup. Does theia provide with any similar to `FrontendApplicationContribution` ? I'm unable to access ApplicationShell in FrontendApllicationContribution. Kindly share some ideas regarding this. 

Thanks.

**Comment by msujew** [comment, marked as answer]
@maheshrajrp You should be able to access the application shell in a `FrontendApplicationContribution` by overriding the `initializeLayout(app: FrontendApplication)` method. You can then access it using `app.shell`.

**Reply by maheshrajrp** [reply]
Hi @msujew ,
I've tried that. That function isn't running. console log doesn't work over there. Kindly assist.

 Thanks

**Reply by maheshrajrp** [reply]
I believe that's because I have previous workbench layout as I reset workbench layout it logs. I need this to run every time.

**Reply by msujew** [reply]
Right, I've mixed it up with `onDidInitializeLayout`. Have you tried that one? That should run every time, instead of just when the layout needs to be initialized for the first time.

**Reply by maheshrajrp** [reply]
Hi @msujew ,
Yes, this works. Thanks !! Didnt realise I could access shell from App itself.
